Add SIG Architecture Code Organization spotlight article

fsmunoz opened the pull request:
SIG Architecture: Code Organization spotlight article.

Closes: #482

fsmunoz commented:
/cc @MadhavJivrajani

nimbinatus commented on content/en/blog/2024/sig-arch-code-spotlight.md:
```diff
@@ -0,0 +1,138 @@
+like to highlight?
+
+**MJ**: I think one very interesting and immensely useful change that has been recently added (and I
+take the opportunity to specifically highlight the work of [Tim Hochin](https://github.com/thockin)
```
```suggestion
take the opportunity to specifically highlight the work of [Tim Hockin](https://github.com/thockin)
```

nimbinatus commented on content/en/blog/2024/sig-arch-code-spotlight.md:
```diff
@@ -0,0 +1,138 @@
+**FSM**: How would someone interested in the topic start helping the subproject?
+
+**MJ**: The first step, as is the first step with any project in Kubernetes, is to join our slack:
+slack.k8s.io, and after that join the `#k8s-code-organization` channel. There is also a
```
Any reason not to make the Slack link an actual link?
```suggestion
[slack.k8s.io](https://slack.k8s.io), and after that join the `#k8s-code-organization` channel. There is also a
```

nimbinatus reviewed (changes requested):
Not going through a detailed grammar review after last time :) Left a couple questions and fixed a spelling error. All links have been checked.

sftim commented on content/en/blog/2024/sig-arch-code-spotlight.md:
```diff
@@ -0,0 +1,138 @@
+**FSM**: Excellent, thank you! Any final comments you would like to share?
+
+**MJ**: The Code Organization subproject always needs help! Especially areas like the publishing
+bot, so don’t hesitate to get involved in the `#k8s-code-organization` slack channel.
```
(nit)
```suggestion
bot, so don’t hesitate to get involved in the `#k8s-code-organization` Slack channel.
```

sftim commented on content/en/blog/2024/sig-arch-code-spotlight.md:
```diff
@@ -0,0 +1,138 @@
+**MJ**: I think one very interesting and immensely useful change that has been recently added (and I
+take the opportunity to specifically highlight the work of [Tim Hockin](https://github.com/thockin)
+on this) is the introduction of [Go workspaces to the Kubernetes
+repo](https://github.com/kubernetes/enhancements/tree/master/keps/sig-architecture/4402-go-workspaces). A
```
Could instead link to [`/blog/2024/03/19/go-workspaces-in-kubernetes/`](https://k8s.dev/blog/2024/03/19/go-workspaces-in-kubernetes/), the article about this.

fsmunoz replied:
Yes much better. Changed in last commit.

sftim commented on content/en/blog/2024/sig-arch-code-spotlight.md:
```diff
@@ -0,0 +1,138 @@
+specific to Kubernetes?
+
+**MJ**: That’s a fair assumption! The first interesting challenge comes from the sheer size of the
+Kubernetes codebase. We have ~2.2 million lines of Go code (which is steadily decreasing thanks to
```
We an use the actual math symbol:
```suggestion
Kubernetes codebase. We have ≅2.2 million lines of Go code (which is steadily decreasing thanks to
```

sftim commented on content/en/blog/2024/sig-arch-code-spotlight.md:
```diff
@@ -0,0 +1,138 @@
+subprojects. We will cover the [SIG Architecture: Code
+Organization](https://github.com/kubernetes/community/blob/master/sig-architecture/README.md#code-organization-1)_
+
+In this SIG Architecture spotlight we talked with [Madhav
```
(nit)
```suggestion
In this SIG Architecture spotlight I talked with [Madhav
```
There were two people talking, one was Madhav, so you'd use first person singular.

fsmunoz replied:
Yes, old habits that are not adequate here.

sftim reviewed (commented):
LGTM apart from publication date.

Very small nits.

/hold
for assignment of publication date.

How about Thursday (2024-04-11)?

fsmunoz commented on content/en/blog/2024/sig-arch-code-spotlight.md:
```diff
@@ -0,0 +1,138 @@
+
+_This is the third interview of a SIG Architecture Spotlight series that will cover the different
+subprojects. We will cover the [SIG Architecture: Code
+Organization](https://github.com/kubernetes/community/blob/master/sig-architecture/README.md#code-organization-1)_
```
Done.

fsmunoz commented:
@nimbinatus @sftim I think this is good to go, I've reflected all suggestions, updated the publication date for the suggested one, and squashed it in case it really is the final version.

nimbinatus commented:
/lgtm

fsmunoz commented:
/unhold

sftim commented:
/lgtm
/approve

sftim commented:
@dims want to send in a PR to do a fixup here too?

(wrt https://github.com/kubernetes/website/pull/45853)

dims commented:
may i defer to @fsmunoz please?

fsmunoz commented:
@dims thanks for the fix, I can add it here if it helps, sure.
